### PR TITLE
GG-34968 Visor check duplicate update counters command

### PR DESCRIPTION
**Pull request by vladErmakov07**

### CODE REVIEW

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/dr/DrCommand.java [hunk 1]
```diff
@@ -108,6 +109,13 @@ public class DrCommand extends AbstractCommand<Object> {
             "<remoteDataCenterId>",
             optional(CMD_AUTO_CONFIRMATION)
         );
+
+        usage(log, "Start partition counters validation for selected caches:",
```
Comment (tledkov-gridgain) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/dr/DrCommand.java:113]: Usage message looks invalid.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/dr/subcommands/DrCheckPartitionCountersCommand.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+         * @return Caches.
+         */
+        public Set<String> caches() {
+            return caches;
+        }
+
+        /**
+         * @return Max number of entries to be checked.
+         */
+        public int checkFirst() {
```
Comment (tledkov-gridgain) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/dr/subcommands/DrCheckPartitionCountersCommand.java:82]: Unused method.

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCheckPartitionCountersTask.java [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Nullable @Override protected VisorDrCheckPartitionCountersTaskResult reduce0(List<ComputeJobResult> results)
+            throws IgniteException {
+        Map<UUID, Collection<VisorDrCheckPartitionCountersJobResult>> nodeMetricsMap = new HashMap<>();
+        Map<UUID, Exception> exceptions = new HashMap<>();
+
+        for (ComputeJobResult res : results) {
+            if (res.getException() != null) {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCheckPartitionCountersTask.java:140]: Please fix codestyle according to 2.x

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCheckPartitionCountersTask.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+                    affectedCaches.addAll(partMetrics.getAffectedCaches());
+                    affectedPartitions.add(part);
+                    brokenEntriesFound += partMetrics.getBrokenEntriesFound();
+                }
+            }
+
+            return new VisorDrCheckPartitionCountersJobResult(cache, size, affectedCaches, affectedPartitions, entriesProcessed, brokenEntriesFound);
+        }
+
+        private DrCachePartitionMetrics calculateForPartition(CacheGroupContext grpCtx, String cache, int part, int checkFirst) {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCheckPartitionCountersTask.java:224]: I cannot catch an idea with check N first entries. What is this feature for? Reduce check workload?
What the user should do when N first entries don't contain errors?
Maybe the flags `findUntilFirstError` or integer parameter `findUntilFirstsError` make sense?

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCachePartitionMetrics.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public long getEntriesProcessed() {
+        return entriesProcessed;
+    }
+
+    public long getBrokenEntriesFound() {
+        return brokenEntriesFound;
+    }
+}
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/visor/dr/VisorDrCachePartitionMetrics.java:64]:
```suggestion
}

```

**Review by AMashenkov — APPROVED** [review]

**Review by AMashenkov — APPROVED** [review]

**Pull request merged**